metrics/options.go

```diff
@@ -79,3 +117,59 @@ func WithErrorHandler(handler errors.ErrorHandler) Option {
 		return nil
 	}
 }
+
+// WithTag sets a tag that will be sent with a specific metric.
+// Parameters:
+//   - k: The tag key
+//   - v: The tag value
+func WithTag(k, v string) MetricOptions {
+	return func(o *metricOptions) error {
+		if k == "" {
+			return fmt.Errorf("tag key cannot be empty")
+		}
+
+		if strings.ContainsAny(k, ":,|=") {
+			return fmt.Errorf("tag key contains invalid characters: %s", k)
+		}
+
+		if strings.ContainsAny(v, ":,|=") {
+			return fmt.Errorf("tag value contains invalid characters: %s", v)
+		}
+
+		if len(k)+len(v)+1 > 200 {
+			return fmt.Errorf("tag %s:%s exceeds maximum length", k, v)
+		}
+
+		for _, reservedTag := range []string{"environment", "service", "version"} {
+			if strings.ToLower(k) == reservedTag {
+				return fmt.Errorf("tag key '%s' is reserved", k)
+			}
+		}
+		o.tags = append(o.tags, fmt.Sprintf(k+":"+v))
+		return nil
+	}
+}
+
+// WithSampleRate sets the sample rate for metrics collection.
+// The sample rate controls what percentage of metrics are actually sent to the backend
+// Parameters:
+//   - rate: A float between 0 and 1 representing the sampling percentage:
+//   - 0: No metrics will be sent (0%)
+//   - 1: All metrics will be sent (100%)
+//   - between 0 and 1 means that % of metrics will be sent (0.25 = 25%)
+//
+// Returns an error if the rate is invalid (negative)
+func WithSampleRate(rate float64) MetricOptions {
+	return func(o *metricOptions) error {
+		if rate < 0 {
+			return fmt.Errorf("sample rate cannot be negative: %f", rate)
+		}
+
+		if rate > 1 {
+			o.sampleRate = 1.0
+			return fmt.Errorf("sample rate %f exceeds maximum of 1.0, capped at 1.0", rate)
+		}
+		o.sampleRate = rate
+		return nil
+	}
+}
```

Comment:
Is is possible to find a way to namespace these so that they do not block or overlap with global function.